l10n_be_cooperator_national_number/models/subscription_request.py

```diff
@@ -0,0 +1,73 @@
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class SubscriptionRequest(models.Model):
+    _inherit = "subscription.request"
+
+    national_number = fields.Char(string="National Number")
+    display_national_number = fields.Boolean(
+        compute="_compute_display_national_number",
+        default=lambda self: self._check_national_number_required())
+
+    @api.depends(
+        "company_id",
+        "company_id.require_national_number",
+        )
+    def _compute_display_national_number(self):
+        self.display_national_number = self._check_national_number_required()
+
+    def _check_national_number_required(self):
+        company = self.env["res.company"]._company_default_get()
+        return company.require_national_number
+
+    def get_national_number_from_partner(self, partner):
+        national_number_id_category = self.env.ref(
+            "l10n_be_national_number.l10n_be_national_number_category"
+            ).id
+        national_number = partner.id_numbers.filtered(
+            lambda rec: rec.category_id.id == national_number_id_category)
+        return national_number.name
+
+    def validate_subscription_request(self):
+        self.ensure_one()
+        if (self._check_national_number_required() and not self.national_number and
+                not self.is_company):
```

Comment:
Also strange indentation : 
```suggestion
        if (
            self._check_national_number_required()
            and not self.national_number
            and not self.is_company
        ):
```